readthedocs/proxito/exceptions.py

```diff
@@ -0,0 +1,110 @@
+"""
+Exceptions for El Proxito.
+
+This module contains exceptions that may be raised in El Proxito application
+but are handled elsewhere in the Django project.
+
+--------------
+404 exceptions
+--------------
+
+Because an Http404 can be raised anywhere in the code,
+we want to add more context for targeted error handling and user-facing communication
+"""
+from django.http.response import Http404
+
+
+class ProxitoHttp404(Http404):
+
+    """General error class for proxity 404s."""
+
+
+class ProxitoProjectHttp404(ProxitoHttp404):
+
+    """Raised if a project was not found."""
+
+    def __init__(self, message, project_slug=None, proxito_path=None):
+        self.project_slug = project_slug
+        self.proxito_path = proxito_path
```

Comment:
I think it should be fine to use just `path` for all these

Reply:
I didn't like the ambiguity around this since we for instance distinguish between a resolved proxito path and the path of the URL. Don't want these two mixed up, i.e. `_proxito_not_found_/path/to/file`

Reply:
Is there a case where we want the non-proxito path? I think we should be fine following the sama names from https://github.com/readthedocs/readthedocs.org/blob/main/readthedocs/proxito/README.rst#url-parts.

Reply:
Maybe `doc_path` would be a better term than just `path`? I wanted to avoid mixing  up the resolved path with the HTTP path (I did that in the first versions of this implementation). In the very nice ascii diagram, things would get confusing if we introduced a URL prefix, i.e. `/en/version/page/` became `/docs/en/version/page/` ?

Reply:
This has also been simplified, it's now an argument that can be added to all contextualized Http404 exceptions, so it's just done once in the new unresolver method in `ServeError404Base`.